### ISSUE TITLE
isValidUntil()

### ISSUE THREAD

**Issue by lioannou** [spatie/ssl-certificate#16]
Hi.

Maybe I'm misunderstanding something - but I cant help but feel `isValidUntil()` is wrong?

The description of the function says:

> Returns true if the certificate is valid and if the expirationDate is **before** the given date.

Shouldnt that be **after** the given date?

i.e. if a certificate expires on 25/10/2017 - then I would have thought `isValidUntil('19/10/2017')` should return `true` - because the certificate is *still valid* on the date of `19/10/2017`.

But currently the package returns the opposite. And the package tests are all opposite as well.

It sounds like this function should really be called `isExpiredOn()` - because that is what the boolean is actually telling you? Or am I missing something here? 

_p.s. I know you need to use `Carbon` objects above - I'm just doing it without for easier reading_

**Comment by mallardduck** [comment]
There's a PR that @lioannou submitted in my fork that addresses this issue. Although, looking over the method, and the issue we're seeing here, I've found another instance where I believe that the current logic provides errant results.

For instance, all of the current tests look _(something)_ like this:

```
        Carbon::setTestNow(Carbon::create('2016', '08', '10', '16', '49', '00', 'utc'));     // 10/08   16:49 - True
        $this->assertTrue($this->certificate->isValidUntil(Carbon::now()->addDays(2)));     // 12/08    16:49 - True

        Carbon::setTestNow(Carbon::create('2016', '08', '10', '16', '49', '00', 'utc'));     // 10/08   16:49 - True
        $this->assertFalse($this->certificate->isValidUntil(Carbon::now()->addDays(8)));      // 18/08  16:49 - False
```
_Sidenote: these tests aren't exact to this repo, these include local changes I'm testing with._

So what if we ad a few tests like this:

```
        Carbon::setTestNow(Carbon::create('2018', '08', '10', '16', '49', '00', 'utc'));     // 10/08/2018   16:49 - False
        $this->assertTrue($this->certificate->isValidUntil(Carbon::create('2016', '08', '12', '16', '49', '00', 'utc')));

        Carbon::setTestNow(Carbon::create('2014', '08', '10', '16', '49', '00', 'utc'));     // 10/08/2014  16:49 - False
        $this->assertTrue($this->certificate->isValidUntil(Carbon::create('2016', '08', '12', '16', '49', '00', 'utc')));
```
Well now we're asking a question that is technically true, but we're getting false. The date provide to the method is within the valid date range.

So we're asking the SSL instance **isValidUntil**, or _are you valid until_, the date I give you? You would expect the answer to be yes, or true, for the date passed but we get false.

**Comment by freekmurze** [comment]
You're right, the function seems to return wrong results.

I've made a fix in the `master` branch. If you don't have any remarks, I'll tag that change.

**Comment by lioannou** [comment]
Cool - looks good. Thanks

**Comment by freekmurze** [comment]
Fix has been tagged `1.3.1`.
